src/app/Processor/ProcessorEdit/ProcessorEdit.tsx

```diff
@@ -45,8 +52,81 @@ const ProcessorEdit = (props: ProcessorEditProps): JSX.Element => {
     type: "",
     parameters: {},
   });
+  const [request, setRequest] = useState<ProcessorRequestData>({
+    name,
+    type: processorType,
+    filters,
+    transformationTemplate: transformation,
+    action,
+    source,
+  });
+  const [isSubmitted, setIsSubmitted] = useState(false);
 
-  const { t } = useTranslation(["openbridgeTempDictionary"]);
+  const {
+    registerValidateConfig,
+    validateName,
+    validation,
+    validate,
+    resetValidation,
+  } = useValidateProcessor(request, existingProcessorName);
+
+  useEffect(() => {
+    setRequest({
+      name,
+      type: processorType,
+      filters,
+      transformationTemplate: transformation,
+      action,
+      source,
+    });
+  }, [name, processorType, filters, transformation, action, source]);
+
+  const handleSubmit = (): void => {
+    setIsSubmitted(true);
+    if (validate() && request) {
+      const requestData: ProcessorRequest = { name: request.name };
+      if (request.type === "sink") {
+        requestData.action = request.action;
+      } else {
+        requestData.source = request.source;
+      }
+      if (request.filters && request.filters.length > 0) {
+        const filtersData = request.filters.filter(
+          (filter) => filter.type && filter.value && filter.key
+        );
+        if (filtersData.length > 0) {
+          requestData.filters =
+            filtersData as unknown as ProcessorRequest["filters"];
+        }
+      }
+      if (
+        request.transformationTemplate &&
+        request.transformationTemplate.trim().length > 0
+      ) {
+        requestData.transformationTemplate =
+          request.transformationTemplate.trim();
+      }
+      onSave(requestData);
+    }
+  };
+
+  useEffect(() => {
+    if (existingProcessorName) {
+      setIsSubmitted(true);
+      validate();
+    }
+  }, [existingProcessorName, validate]);
+
+  useEffect(() => {
+    if (isSubmitted) {
+      document.querySelector(".processor-field-error")?.scrollIntoView({
```

Comment:
I suggest dynamically putting (`processor-field-error`) in a React ref, and calling `scrollIntoView` on it, instead of querying the document.

Reply:
You caught me. The issue I was facing is that I'm not scrolling to a single specific node. I need to target the first input field in the page with validation issues, wherever it is. 

Doing it with refs means registering and un-registering the refs of fields with the order they appear on page, so to know which one to scroll to.

I took the shortcut of querying directly the document because it's way more easier, even if it's not the canonical react way of doing this things.